lib/improver/tests/utilities/test_DifferenceBetweenAdjacentGridSquares.py

```diff
@@ -191,6 +202,33 @@ def test_masked_data(self):
         self.assertArrayEqual(result.data.mask, expected.mask)
 
 
+class Test_gradient_from_diff(IrisTest):
+
+    """Test for correct behaviour when calculating a gradient"""
+
+    def setUp(self):
+        """Set up cube."""
+        data = np.array([[1, 2, 3],
+                         [2, 4, 6],
+                         [5, 10, 15]])
+        x_coord = DimCoord(2.*np.arange(3), "projection_x_coordinate")
+        y_coord = DimCoord(2.*np.arange(3), "projection_y_coordinate")
+        self.cube = iris.cube.Cube(data, "wind_speed", units="m s-1",
+                                   dim_coords_and_dims=[(y_coord, 0),
+                                                        (x_coord, 1)])
+        self.plugin = DifferenceBetweenAdjacentGridSquares(gradient=True)
+
+    def test_basic(self):
+        """Test contents and metadata"""
+        expected = np.array([[0.5, 0.5, 0.5],
+                             [1.0, 1.0, 1.0],
+                             [2.5, 2.5, 2.5]])
+        xdiff = self.plugin.calculate_difference(self.cube, "x")
+        result = self.plugin.gradient_from_diff(xdiff, self.cube, "x")
+        self.assertArrayAlmostEqual(result.data, expected)
+        self.assertEqual(result.name(), "gradient_of_wind_speed")
+
```

Comment:
Again, see comments later. Can we have a negative gradient somewhere in this test.